lib/posix/pthread.c

```diff
@@ -667,15 +667,15 @@ int pthread_join(pthread_t pthread, void **status)
 	{
 		if (t->detachstate != PTHREAD_CREATE_JOINABLE) {
 			ret = EINVAL;
-			K_SPINLOCK_BREAK;
 			LOG_ERR("Pthread %p is not a joinable", &t->thread);
+			K_SPINLOCK_BREAK;
```

Comment:
Ack yeah, good catch. It's probably not great to log with the spinlock held though.

Is it possible to defer the call to log_err ? Maybe just by using a `const char *` that is initialized to `NULL`?

Reply:
Maybe remove these logs at all?

Reply:
Sure

Reply:
> Ack yeah, good catch. It's probably not great to log with the spinlock held though.
> 
> Is it possible to defer the call to log_err ? Maybe just by using a `const char *` that is initialized to `NULL`?

like this? #0901f63